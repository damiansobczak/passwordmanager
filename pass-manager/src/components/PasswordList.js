import React from "react";
import PasswordItem from "./PasswordItem";

class PasswordList extends React.Component {
  render() {
    return (
      <div className="card">
        <div className="card-content">
          <div className="card-title">Twoje has≈Ça</div>
          <PasswordItem />
        </div>
      </div>
    );
  }
}

export default PasswordList;
