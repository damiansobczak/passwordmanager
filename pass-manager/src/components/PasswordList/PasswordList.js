import React from "react";
import PasswordItem from "../PasswordItem/PasswordItem";

class PasswordList extends React.Component {
  render() {
    return (
      <div className="card">
        <div className="card-content">
          <div className="card-title">Twoje has≈Ça</div>
          {this.props.passwords.map((password, index) => (
            <PasswordItem name={password.name} hash={password.password} date={password.date} key={index} />
          ))}
        </div>
      </div>
    );
  }
}

export default PasswordList;
