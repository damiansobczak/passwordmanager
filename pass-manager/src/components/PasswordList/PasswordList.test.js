import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import PasswordList from "./PasswordList";

describe("<PasswordList />", () => {
  const passwords = [{ name: "Google", password: "123dsadasd", date: "12-04-2019" }, { name: "Youtube", password: "wwwwwsda", date: "14-05-2019" }, { name: "Facebook", password: "111111111111", date: "17-06-2019" }];
  it("renders correctly", () => {
    const tree = renderer.create(<PasswordList passwords={passwords} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders password items", () => {
    const wrapper = shallow(<PasswordList passwords={passwords} />);
    expect(wrapper.find("PasswordItem")).toHaveLength(3);
  });
});
