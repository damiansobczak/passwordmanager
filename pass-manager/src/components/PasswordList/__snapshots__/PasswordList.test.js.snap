// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordList /> renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="card-title"
    >
      Twoje hasła
    </div>
    <div
      className="password"
    >
      <div
        className="password-name"
      >
        Google
      </div>
      <input
        className="password-hash"
        defaultValue="123dsadasd"
        readOnly={true}
        type="password"
      />
      <div
        className="password-date"
      >
        12-04-2019
      </div>
      <div
        className="password-actions"
      >
        <div
          className="white red-text waves-effect waves-teal btn-flat password-reveal"
          onClick={[Function]}
        >
          Odsłoń
        </div>
        <div
          className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy"
          onClick={[Function]}
        >
          Kopiuj
        </div>
      </div>
    </div>
    <div
      className="password"
    >
      <div
        className="password-name"
      >
        Youtube
      </div>
      <input
        className="password-hash"
        defaultValue="wwwwwsda"
        readOnly={true}
        type="password"
      />
      <div
        className="password-date"
      >
        14-05-2019
      </div>
      <div
        className="password-actions"
      >
        <div
          className="white red-text waves-effect waves-teal btn-flat password-reveal"
          onClick={[Function]}
        >
          Odsłoń
        </div>
        <div
          className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy"
          onClick={[Function]}
        >
          Kopiuj
        </div>
      </div>
    </div>
    <div
      className="password"
    >
      <div
        className="password-name"
      >
        Facebook
      </div>
      <input
        className="password-hash"
        defaultValue="111111111111"
        readOnly={true}
        type="password"
      />
      <div
        className="password-date"
      >
        17-06-2019
      </div>
      <div
        className="password-actions"
      >
        <div
          className="white red-text waves-effect waves-teal btn-flat password-reveal"
          onClick={[Function]}
        >
          Odsłoń
        </div>
        <div
          className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy"
          onClick={[Function]}
        >
          Kopiuj
        </div>
      </div>
    </div>
  </div>
</div>
`;
